extra/samples/stduart/main.cpp

```diff
@@ -121,7 +123,8 @@ int main(int argc, char** argv) {
         }
         if (std::cin.eof())
             break;
-        uart0.rx().write((const char[]){'\n'});
+        std::span<const char> newline{{'\n'}};
```

Comment:
This should be `constexpr`, and because we will want to move away from `std::span` (the libSMCE interface is supposed to be C++17, but `std::span` use in BoardView for UART buffers effectively bumps that to C++20), we might as well use `std::array` instead of `std::span`

Reply:
In fact, with `std::array`, the temporary variable could probably even be elided

Reply:
Would `uart0.rx().write({{'\n'}})` be a possible option?

Reply:
If it's correct C++17 and works on all supported compilers, sure